src/server/cluster/outgoing_slot_migration.cc

```diff
@@ -70,7 +76,7 @@ MigrationState OutgoingMigration::GetState() const {
 }
 
 MigrationState OutgoingMigration::GetStateImpl() const {
-  MigrationState min_state = MigrationState::C_STABLE_SYNC;
+  MigrationState min_state = MigrationState::C_FINISHED;
```

Comment:
This implementation can lead to bugs as no one would expect finished must be the last in the enum
I suggest add another enum for max invalid